extensions/amp-subscriptions/0.1/amp-subscriptions.js

```diff
@@ -257,6 +273,45 @@ export class SubscriptionService {
     return this;
   }
 
+  /**
+   * Delegates authentication to viewer
+   */
+  delegateAuthToViewer_() {
+    const serviceIds = ['local'];
+    const currentProductId = /** @type {string} */ (user().assert(
+        this.pageConfig_.getProductId(),
+        'Product id is null'
+    ));
+    const publicationId = /** @type {string} */ (user().assert(
+        this.pageConfig_.getPublicationId(),
+        'Publication id is null'
+    ));
+    const origin = getWinOrigin(this.ampdoc_.win);
+    this.platformStore_ = new PlatformStore(serviceIds);
+    this.viewer_.sendMessageAwaitResponse('auth', dict({
+      'publicationId': publicationId,
+      'productId': currentProductId,
+      'origin': origin,
+    })).then(entitlementData => {
+      const authData = (entitlementData || {})['authorization'];
+      dev().assert(authData, 'authorization is not defined');
+      const decodedData = this.jwtHelper_.decode(authData);
+      if (decodedData['aud'] != origin) {
+        user().error(TAG, 'Audience does not match');
+      }
+      if (Date.now() - decodedData['exp'] < 1000 * 60) {
```

Comment:
That's reverse, right? The `exp` should be in the future, e.g. `exp > now => no error`. No? Otherwise, strict check is enough. It's the job of the viewer to give us the token with some buffer.